brooklyn-tosca-transformer/src/main/java/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaNodeToEntityConverter.java

```diff
@@ -355,9 +386,73 @@ protected void applyLifecycle(Map<String, Operation> ops, String opKey, EntitySp
 
     }
 
+    protected void configureRuntimeEnvironment(EntitySpec<?> entitySpec) {
+        if (indexedNodeTemplate.getArtifacts() == null) {
+            return;
+        }
+
+        final Map<String, String> filesToCopy = MutableMap.of();
+        final List<String> preInstallCommands = MutableList.of();
+        final List<String> envCommands = MutableList.of();
+
+        for (final Map.Entry<String, DeploymentArtifact> artifactEntry : indexedNodeTemplate.getArtifacts().entrySet()) {
+            if (artifactEntry.getValue() == null) {
+                continue;
+            }
+            if (!"tosca.artifacts.File".equals(artifactEntry.getValue().getArtifactType())) {
+                continue;
+            }
+
+            final String destRoot = Os.mergePaths("~", artifactEntry.getValue().getArtifactName());
+            final String tempRoot = Os.mergePaths("/tmp", artifactEntry.getValue().getArtifactName());
+
+            preInstallCommands.add("mkdir -p " + destRoot);
```

Comment:
Suggest putting the files in a special directory like "brooklyn-tosca-resources" (or whatever) rather than just dumping them in the home directory.